ztp/policygenerator/policyGen/policyBuilder_test.go

```diff
@@ -1362,3 +1362,120 @@ spec:
 		assert.NotNil(t, policies)
 	}
 }
+
+func TestNamespaceInvalidEvaluationInterval(t *testing.T) {
+	tests := []struct {
+		input         string
+		expectedError string
+	}{{
+		// Spec compliant interval invalid time unit
+		input: `
+apiVersion: ran.openshift.io/v1
+kind: PolicyGenTemplate
+metadata:
+  name: "test"
+  namespace: "test"
+spec:
+  bindingRules:
+    justfortest: "true"
+  evaluationInterval:
+    compliant: 30min
+    noncompliant: 20s
+  sourceFiles:
+    - fileName: GenericNamespace.yaml
+      policyName: "gen-policy"
+    - fileName: GenericSubscription.yaml
+      policyName: "gen-policy"
+`,
+		expectedError: `evaluationInterval.compliant 'time: unknown unit "min" in duration "30min"'`,
+	}, {
+		// Spec noncompliant interval missing time unit
+		input: `
+apiVersion: ran.openshift.io/v1
+kind: PolicyGenTemplate
+metadata:
+  name: "test"
+  namespace: "test"
+spec:
+  bindingRules:
+    justfortest: "true"
+  evaluationInterval:
+    compliant: 20m
+    noncompliant: 20
+  sourceFiles:
+    - fileName: GenericNamespace.yaml
+      policyName: "gen-policy"
+    - fileName: GenericSubscription.yaml
+      policyName: "gen-policy"
+`,
+		expectedError: `evaluationInterval.noncompliant 'time: missing unit in duration "20"'`,
+	}, {
+		// sourceFile level compliant interval invalid time unit
+		input: `
+apiVersion: ran.openshift.io/v1
+kind: PolicyGenTemplate
+metadata:
+  name: "test"
+  namespace: "test"
+spec:
+  bindingRules:
+    justfortest: "true"
+  evaluationInterval:
+    compliant: 20m
+    noncompliant: 20s
+  sourceFiles:
+    - fileName: GenericNamespace.yaml
+      policyName: "gen-policy"
+    - fileName: GenericSubscription.yaml
+      policyName: "gen-policy"
+      evaluationInterval:
+        compliant: 1hour
+`,
```

Comment:
It would be good to verify the check on nonCompliant under sourceFiles as well.
I would also be good to add a test case which shows it passing when valid inputs are given in each of the levels (one test can validate all 4)

Reply:
The nonCompliant under sourceFiles is checked on Line 1459.  I added a test for valid inputs.